llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1560,6 +1560,13 @@ def combine_shuffle_concat : GICombineRule<
         [{ return Helper.matchCombineShuffleConcat(*${root}, ${matchinfo}); }]),
   (apply [{ Helper.applyCombineShuffleConcat(*${root}, ${matchinfo}); }])>;
 
+// Combines shuffles of vector into extract_subvector
+def combine_shuffle_vector : GICombineRule<
```

Comment:
Name is too broad, and doesn't match the comment. It says it forms an extract_subvector, but this is creating a build_vector

Reply:
updated to `combine_shuffle_vector_to_build_vector`